test/signalrclient-testhost/Connections/RawConnection.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using System.Threading;
+using Microsoft.AspNet.SignalR;
+using Newtonsoft.Json;
+
+namespace SelfHost
+{
+    public class RawConnection : PersistentConnection
+    {
+        protected override Task OnReceived(IRequest request, string connectionId, string data)
+        {
+            var message = JsonConvert.DeserializeObject<Message>(data);
+
+            switch (message.Type)
```

Comment:
do you have plans to build on it? because now it is an overkill... why not just send back `message.Value,ToString()` + `message.Type.ToString()`?